src/qt/walletframe.h

```diff
@@ -36,7 +36,7 @@ class WalletFrame : public QFrame
 
     void setClientModel(ClientModel *clientModel);
 
-    void addWallet(WalletModel *walletModel);
+    bool addWallet(WalletModel *walletModel);
```

Comment:
I would prefer a `bool ready()` method that simply checks the conditions rather than arbitrarily booling non-bool methods like this. I think it's unnecessarily confusing.

Reply:
6d6a7a8403ae923f189812edebdd95761de0e7f2

@kallewoof thanks for reviewing.

> arbitrarily booling non-bool methods like this.

What you mean "like this"? I could refactor by adding `bool WalletFrame::hasWallet(WalletModel*) const` but IMO this is also fine, for instance, `std::set::insert` also returns whether the item was added.

What it could have is the `NODISCARD` attribute, but ATM it's not used in src/qt.

Reply:
IMO it's an anti-pattern to silently fail when an operation is attempted, returning a success bool instead makes sense to me in that case, or alternatively raise on failure and use `ready` as you suggest.

Reply:
Yeah, I know this is done elsewhere (even in `std`), but I don't think it's a good idea, personally. It often results in having to look up what exactly is being returned, made even worse with things like `bool CreateThing(...)`. (Looking at the signature unless `Thing` is a bool type, the return value is most likely the result of the create operation, but seeing it in the wild as `auto foo = CreateThing(...)` you won't know if foo is a bool or a `Thing` without checking.)

Agree that failing silently is a bad idea. Raising sounds good to me, unless you really like the bool approach, in which case keep it as is.